src/pages/SettingsPage.js

```diff
@@ -0,0 +1,172 @@
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+import React, { useCallback, useState, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import ValidUrl from 'valid-url';
+import { Text, View, ToastAndroid, TouchableOpacity } from 'react-native';
+import { hashPassword } from 'sussol-utilities';
+import { Button } from 'react-native-ui-components';
+
+import { UIDatabase } from '../database';
+import { SETTINGS_KEYS } from '../settings';
+import { MODAL_KEYS } from '../utilities';
+
+import { gotoRealmExplorer } from '../navigation/actions';
+
+import { ConfirmIcon } from '../widgets/icons';
+import { DataTablePageView, PageInfo } from '../widgets/index';
+import { DataTablePageModal } from '../widgets/modals/index';
+
+import globalStyles from '../globalStyles';
+
+const exportData = async () => {
+  const syncSiteName = UIDatabase.getSetting(SETTINGS_KEYS.SYNC_SITE_NAME);
+  const { success, message } = await UIDatabase.exportData(syncSiteName);
+  const toastMessage = success ? 'Exported data file' : `Couldn't export data: ${message}`;
+  ToastAndroid.show(toastMessage, ToastAndroid.SHORT);
+};
+
+const Settings = ({ toRealmExplorer, currentUser }) => {
+  const [state, setState] = useState({
+    syncURL: UIDatabase.getSetting(SETTINGS_KEYS.SYNC_URL),
+    modalKey: '',
+    syncPassword: '',
```

Comment:
I think because of this, when you just edit the URL and save you'll wipe the password to `''`, which is oof.

Actually you wouldn't even have to save the URL, just save without doing anything and kill the password.

Reply:
e87efd0cddc7eb2b66d1d5124a38d2c6b6e0ebd4